Apromore-Custom-Plugins/Process-Discoverer-Portal-Plugin/src/main/java/org/apromore/plugin/portal/processdiscoverer/PDController.java

```diff
@@ -174,15 +174,21 @@ public ResourceBundle getLabels() {
     }
 
     public String getLabel(String key) {
-        String label = getLabels().getString(key);
+        String label;
+        label = getLabel(key, null);
         if (label == null) {
-            label = "";
+            label = key;
         }
         return label;
     }
 
     public String getLabel(String key, String defaultVal) {
-        String label = getLabels().getString(key);
+        String label;
+        try {
+            label = getLabels().getString(key);
+        } catch (MissingResourceException e) {
+            label = null;
+        }
```

Comment:
Can we replace the method code with:
try {
    return getLabels().getString(key);
}
catch (Exception) {
    return defaultVal;
}